src/catchup/DownloadApplyTxsWork.cpp

```diff
@@ -43,44 +43,40 @@ DownloadApplyTxsWork::yieldMoreWork()
     {
         throw std::runtime_error("Work has no more children to iterate over!");
     }
+    std::vector<FileType> fileTypesToDownload{
+        FileType::HISTORY_FILE_TYPE_TRANSACTIONS};
+    std::vector<std::shared_ptr<BasicWork>> downloadSeq;
+    std::vector<FileTransferInfo> filesToTransfer;
+    if (mApp.getConfig().CATCHUP_SKIP_KNOWN_RESULTS)
+    {
+        fileTypesToDownload.emplace_back(FileType::HISTORY_FILE_TYPE_RESULTS);
+    }
+    for (auto const& fileType : fileTypesToDownload)
```

Comment:
nit: code would be easier to follow if you kept the `getAndUnzip` work creation as is, and appended results work to the sequence separately behind the CATCHUP_SKIP_KNOWN_RESULTS flag (it's only two file types, so I think the loop can be avoided)